sci-geosciences/josm-bin/josm-bin-18004.ebuild

```diff
@@ -0,0 +1,32 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit java-utils-2 desktop
+
+DESCRIPTION="Java-based editor for the OpenStreetMap project"
+HOMEPAGE="https://josm.openstreetmap.de/"
+SRC_URI="https://josm.openstreetmap.de/download/josm-snapshot-${PV}.jar"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND=">=virtual/jre-1.8"
+
+S="${WORKDIR}"
+
+src_install() {
+	java-pkg_newjar "${DISTDIR}/${A}" "${PN}.jar" || die "java-pkg_dojar failed"
+	java-pkg_dolauncher "${PN}" --jar ${PN}.jar || die "java-pkg_dolauncher failed"
```

Comment:
Not familiar with these functions, but they do already handle calling `|| die` internally so that bit isn't needed.

`${PN}` is also guaranteed to have no spaces, only `${DISTDIR}` needs quotes here.